### PR TITLE
feat(density): Add density subsystem to components

### PR DESCRIPTION
**Pull request by abhiomkar**
### Description

The styles for applying density to button component instance looks like this:

```scss
@import "@material/button/mixins";

.my-custom-button {
  @include mdc-button-density(-3);
}
```

This would apply `-3` (high density) to button component instance. Similar to shape subsystem every component will provide a density mixin which is used to adjust the visual height without breaking the structure & animations if any.

#### Components that implements density in this PR

* Button
* Checkbox
* Data Table

Refs #4889 

BREAKING CHANGE: Renamed sass mixins & variables in MDC Data Table - `mdc-data-table-header-row-height` => `mdc-data-table-header-cell-height` & `mdc-data-table-row-height` => `mdc-data-table-cell-height`. Also removed `mdc-button--dense` variant, use button's density mixin instead.

### CODE REVIEW

File: packages/mdc-button/_mixins.scss [hunk 1]
```diff
@@ -227,27 +228,48 @@ $mdc-button-ripple-target: ".mdc-button__ripple";
...
+
+  @include mdc-feature-targets($feat-structure) {
+    height: $height;
   }
+}
+
+@mixin mdc-button-shape-radius($radius,
+$rtl-reflexive: false,
+$density-scale: $mdc-button-density-default-scale,
+$query: mdc-feature-all()) {
```
Comment (joyzhong) [inline packages/mdc-button/_mixins.scss]: nit: Indent +2

File: packages/mdc-button/package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "@material/rtl": "^3.1.0",
     "@material/shape": "^3.1.0",
     "@material/theme": "^3.1.0",
-    "@material/typography": "^3.1.0"
+    "@material/typography": "^3.1.0",
+    "@material/density": "^0.0.0"
```
Comment (joyzhong) [inline packages/mdc-button/package.json]: Alphabetize

File: packages/mdc-checkbox/_mixins.scss [hunk 3]
```diff
@@ -26,6 +26,7 @@
 @import "@material/ripple/mixins";
 @import "@material/rtl/mixins";
 @import "@material/touch-target/mixins";
+@import "@material/density/functions";
```
Comment (joyzhong) [inline packages/mdc-checkbox/_mixins.scss]: Alphabetize

File: packages/mdc-button/_mixins.scss (hunk 1 above)
Comment (abhiomkar) [reply packages/mdc-button/_mixins.scss]: image

Stylelint throws error for +2 indentation on params.

**Pull request merged**